### PR TITLE
[backport v1.0.3] Support topology aware scheduling of guest cluster workloads

### PR DESCRIPTION
**Pull request by chrisho**
backport PR:
https://github.com/harvester/harvester/pull/2428

Related Issue:
https://github.com/harvester/harvester/issues/1418

### CODE REVIEW

**Comment by guangbochen** [comment]
The commits are incorrect and u should use cherry-pick instead. Also, the backport PR link is not correct it should be https://github.com/harvester/harvester/pull/2428.

**Comment by chrisho** [comment]
> The commits are incorrect and u should use cherry-pick instead. Also, the backport PR link is not correct it should be #2428.

updated

**Review by futuretea — APPROVED** [review]
LGTM

File: pkg/controller/master/virtualmachine/vmi_controller.go [hunk 1]
```diff
@@ -91,3 +105,41 @@ func (h *VMIController) UnsetOwnerOfPVCs(_ string, vmi *kubevirtv1.VirtualMachin
...
+	if creator := vmi.Labels[builder.LabelKeyVirtualMachineCreator]; creator != VirtualMachineCreatorNodeDriver {
+		return vmi, nil
+	}
+
+	node, err := h.nodeCache.Get(vmi.Status.NodeName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return vmi, nil
+		}
+		return vmi, fmt.Errorf("failed to get node %s for VirtualMachineInstance %s: %v", vmi.Status.NodeName, vmi.Name, err)
```
Comment (w13915984028) [inline pkg/controller/master/virtualmachine/vmi_controller.go:124]: Better to add a log when error returning here.

Comment (chrisho) [reply pkg/controller/master/virtualmachine/vmi_controller.go:124]: ~updated~

Comment (chrisho) [reply pkg/controller/master/virtualmachine/vmi_controller.go:124]: When the controller returns an error, the framework will print the error, so we don't need another output log.
cc @guangbochen

Comment (guangbochen) [reply pkg/controller/master/virtualmachine/vmi_controller.go:124]: Just to emphasize here that we don't change the code in the backport PR, if needed feel free to file a new PR to make an enhancement, thanks.

**Review by guangbochen — APPROVED** [review]

**Pull request merged**